src/execution/execute.js

```diff
@@ -1027,9 +1038,10 @@ function completeAbstractValue(
   path: ResponsePath,
   result: mixed,
 ): MaybePromise<ObjMap<mixed>> {
-  const runtimeType = returnType.resolveType
-    ? returnType.resolveType(result, exeContext.contextValue, info)
-    : defaultResolveTypeFn(result, exeContext.contextValue, info, returnType);
+  const resolveTypeFn = returnType.resolveType || exeContext.typeResolver;
+  const contextValue = exeContext.contextValue;
+  const typeInfo = { ...info, valueType: returnType, valuePath: path };
+  const runtimeType = resolveTypeFn(result, contextValue, typeInfo);
```

Comment:
Can we avoid creating a new object for every field resolution?

Perhaps these lines should be:

```
const runtimeType = returnType.resolveType
  ? returnType.resolveType(result, exeContext.contextValue, info)
  : exeContext.typeResolver
    ? exeContext.typeResolver(returnType, result, exeContext.contextValue, info)
    : defaultResolveTypeFn(returnType, result, exeContext.contextValue, info)
```

Reply:
Open to this type signature - I'm not sure how valuable `returnType` will be for a default resolver - if it deserves to be the first argument or not. I imagine `result` would still be the most useful.

Another option might be to just provide `returnType` as another argument after info to all type resolvers - not just default ones?